lib/checkother.cpp

```diff
@@ -4364,7 +4364,8 @@ void CheckOther::overlappingWriteUnion(const Token *tok)
 
 void CheckOther::overlappingWriteFunction(const Token *tok)
 {
-    const std::string &funcname = tok ? tok->str() : emptyString;
```

Comment:
Seems like this overlaps with the intent of https://github.com/danmar/cppcheck/pull/7251 (should just use `""`)

Reply:
Good catch. That should never be `nullptr` otherwise the error message would have an empty function name. I will align it to match existing code and integrate with the other PR.

Reply:
I assume that code exists for the `getErrorMessages()` invocation.

Reply:
Done.